src/dropdown/dropdown.ts

```diff
@@ -205,22 +214,17 @@ export class NgbDropdown implements OnInit,
     }
   }
 
-  closeFromClick($event) {
-    if (this.autoClose && $event.button !== 2 && !this._isEventFromToggle($event)) {
+  private _shouldCloseFromClick(event: MouseEvent) {
+    if (event.button !== 2 && !this._isEventFromToggle(event)) {
       if (this.autoClose === true) {
-        this.close();
-      } else if (this.autoClose === 'inside' && this._isEventFromMenu($event)) {
-        this.close();
-      } else if (this.autoClose === 'outside' && !this._isEventFromMenu($event)) {
-        this.close();
+        return true;
+      } else if (this.autoClose === 'inside' && this._isEventFromMenu(event)) {
+        return true;
+      } else if (this.autoClose === 'outside' && !this._isEventFromMenu(event)) {
+        return true
       }
     }
-  }
-
-  closeFromOutsideEsc() {
-    if (this.autoClose) {
```

Comment:
Is there any specific reason you removed completely this check (`if (this.autoClose)`)  ?

Reply:
this is checked now when setting handlers in `_setCloseHandlers()`:

https://github.com/ng-bootstrap/ng-bootstrap/pull/2518/files#diff-ed4f415f7194ca32cfe9021c01b9badfR179